### PR TITLE
Add with-joi example

### PR DESCRIPTION
**Pull request by sergioalvz**
Hi there! 👋  

This pull request adds a new example to show how to use [`joi`](https://github.com/sideway/joi) and [`next-joi`](https://github.com/codecoolture/next-joi) to validate Next.js API Routes.

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

**Comment by sergioalvz** [comment]
Hello, folks! 

It has been around 2 months that I opened this pull request. Any help so I can move it forward?

File: examples/with-joi/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "with-joi",
+  "version": "0.1.0",
```
Comment (leerob) [inline examples/with-joi/package.json:3]:
```suggestion
```

File: examples/with-joi/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "joi": "17.4.0",
+    "next": "10.2.3",
```
Comment (leerob) [inline examples/with-joi/package.json:12]: Could you update all deps to latest?

Comment (sergioalvz) [reply examples/with-joi/package.json]: Thanks for the review, @leerob! I did the change.

**Review by leerob — APPROVED** [review]

**Pull request merged**